### PR TITLE
Positional args

### LINKED ISSUE #26: Support positional arguments

### PR DESCRIPTION
**Pull request by rogiervandergeer**
Closes #26

### CODE REVIEW

File: src/pydargs/__init__.py [hunk 1]
```diff
@@ -46,101 +46,102 @@ def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
         if field.metadata.get("ignore_arg", False):
             continue
 
+        kwargs: dict[str, Any] = dict()
         field_has_default = field.default is not MISSING or field.default_factory is not MISSING
+        if positional := field.metadata.get("positional", False):
```
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py]:
```suggestion
        if field.metadata.get("positional", False) is True:
```

File: src/pydargs/__init__.py [hunk 2]
```diff
@@ -46,101 +46,102 @@ def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
         if field.metadata.get("ignore_arg", False):
```
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py:46]: are we adding the positionals in the order of the fields? is `fields` always ordered in the same way?

File: src/pydargs/__init__.py [hunk 3]
```diff
@@ -46,101 +46,102 @@ def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
...
-                dest=field.name,
                 help=f"Override field {field.name}.",
-                required=field.default is MISSING and field.default_factory is MISSING,
                 type=parser_fct,
+                **kwargs,
             )
         elif origin := get_origin(field.type):
             if origin is Sequence or origin is list:
-                required = field.default is MISSING and field.default_factory is MISSING
+                kwargs["nargs"] = "+" if not field_has_default else "*"
```
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py]:
```suggestion
                kwargs["nargs"] = "*" if field_has_default else "+"
```

File: src/pydargs/__init__.py [hunk 4]
```diff
@@ -46,101 +46,102 @@ def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
...
-                required=not field_has_default,
                 type=partial(
                     _parse_datetime, is_date=field.type is date, date_format=field.metadata.get("date_format")
                 ),
+                **kwargs,
             )
         elif field.type is bool:
             if field.metadata.get("as_flags", False):
+                if positional:
+                    raise ValueError()
```
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py]: can we explain why this is causing a value error in the message?
```suggestion
                    raise ValueError("Positional arguments can not serve as flags")
```

File: tests/test_types.py [hunk 5]
```diff
@@ -11,45 +11,55 @@
 class TestBase:
     @dataclass
     class Config:
-        a: int
+        a: int = field(metadata=dict(positional=True))
         b: str
         c: float = 1.0
         d: int = 4
-        e: str = "e"
+        e: str = field(metadata=dict(positional=True), default="e")
```
Comment (antonsteenvoorden) [inline tests/test_types.py:18]:
```suggestion
        e: str = field(default="e", metadata=dict(positional=True))
```

**Review by antonsteenvoorden — CHANGES_REQUESTED** [review]
minor changes

File: src/pydargs/__init__.py (hunk 4 above)
Comment (antonsteenvoorden) [reply src/pydargs/__init__.py]: This is just an example btw, I'm sure you can come up with something better 😜

File: src/pydargs/__init__.py (hunk 2 above)
Comment (rogiervandergeer) [reply src/pydargs/__init__.py:46]: Fields are ordered in the order of definition inside the dataclass. So we are adding them in that order.

File: src/pydargs/__init__.py (hunk 1 above)
Comment (rogiervandergeer) [reply src/pydargs/__init__.py]: We need `positional` down below.

File: src/pydargs/__init__.py (hunk 4 above)
Comment (rogiervandergeer) [reply src/pydargs/__init__.py]:
```suggestion
                    raise ValueError("A field cannot be positional as well as be represented by flags.")
```
Something like this?

File: tests/test_types.py (hunk 5 above)
Comment (rogiervandergeer) [reply tests/test_types.py:18]: Does this matter? I think it has a little value for testing purposes

File: src/pydargs/__init__.py (hunk 1 above)
Comment (antonsteenvoorden) [inline src/pydargs/__init__.py]:
```suggestion
        positional = field.metadata.get("positional", False)
        if positional:
```

File: src/pydargs/__init__.py (hunk 4 above)
Comment (antonsteenvoorden) [reply src/pydargs/__init__.py]: 👍

**Review by antonsteenvoorden — APPROVED** [review]
Nice!

**Pull request merged**